export/platform/support/freshdesk/main.py

```diff
@@ -0,0 +1,222 @@
+"""
+Lambda function to fetch Freshdesk tickets that have been created
+or updated in the past day and save them to S3
+"""
+
+import base64
+import json
+import logging
+import os
+
+from datetime import datetime, timedelta
+
+import boto3
+import requests
+
+logger = logging.getLogger()
+logger.setLevel("INFO")
+
+# Env vars
+FRESHDESK_DOMAIN = os.environ.get("FRESHDESK_DOMAIN")
+FRESHDESK_API_KEY_PARAMETER_NAME = os.environ.get("FRESHDESK_API_KEY_PARAMETER_NAME")
+S3_BUCKET_NAME = os.environ.get("S3_BUCKET_NAME")
+S3_OBJECT_PREFIX = os.environ.get("S3_OBJECT_PREFIX")
+
+# Label lookups
+SOURCE_LOOKUP = {
+    1: "Email",
+    2: "Portal",
+    3: "Phone",
+    7: "Chat",
+    9: "Feedback Widget",
+    10: "Outbound Email",
+}
+
+STATUS_LOOKUP = {
+    2: "Open",
+    3: "Pending",
+    4: "Resolved",
+    5: "Closed",
+    6: "Waiting on Customer",
+    7: "Waiting on Third Party",
+}
+
+PRIORITY_LOOKUP = {1: "Low", 2: "Medium", 3: "High", 4: "Urgent"}
+
+CONVERSATION_SOURCE = {
+    0: "Reply",
+    2: "Note",
+}
+
+
+class FreshdeskClient:
+    """Client to interact with Freshdesk API"""
+
+    def __init__(self, domain, api_key):
+        """Initialize client with domain, api_key and headers"""
+        self.base_url = f"https://{domain}"
+        self.encoded_key = base64.b64encode(f"{api_key}:X".encode()).decode()
+        self.headers = {
+            "Authorization": f"Basic {self.encoded_key}",
+            "Content-Type": "application/json",
+        }
+        self.products_cache = self.get_products()
+        self.contacts_cache = {}
+
+    def get_products(self):
+        logger.info("Fetching products from API...")
+        products = {}
+        url = f"{self.base_url}/api/v2/products"
+        response = requests.get(url, headers=self.headers)
+        if response.status_code == 200:
+            products = {
+                str(product["id"]): product["name"] for product in response.json()
+            }
+        else:
+            logger.info(f"Error fetching products: {response.status_code}")
+        return products
+
+    def get_ticket_conversations(self, ticket_id):
+        """Fetch all ticket conversations"""
+        logger.info(f"Fetching conversations from API for {ticket_id}...")
+        page = 1
+        per_page = 100
+        reply_count = 0
+        note_count = 0
+        while page <= 10:  # Limit to 1000 conversations
+            url = f"{self.base_url}/api/v2/tickets/{ticket_id}/conversations?per_page={per_page}&page={page}"
+            response = requests.get(url, headers=self.headers)
+            if response.status_code == 200:
+                conversations = response.json()
+                reply_count += len([c for c in conversations if c.get("source") == 0])
```

Comment:
Could we use the lookup table for the source to make this line easier to read?

Reply:
For sure - I've updated the code to check that value of a constant here